### DISCUSSION TITLE
Feature Request: UPI support.

### DISCUSSION THREAD

**Discussion by khitab** [oss-apps/split-pro#315] [Ideas]
Having UPI integration in the app will make the Settle Up part much easier.

**Flow:**

1. Users can add their UPI IDs to the profile.
2. Clicking on Settle Up using UPI would open installed UPI app, or in case of web, a QR code.

**Reference:**

- [How to Generate Dynamic QR Codes to Collect Payments through UPI](https://www.labnol.org/dynamic-upi-qr-codes-220426)
- [UPI Deeplinks specs](https://www.npci.org.in/PDF/npci/upi/circular/2017/Circular18_BankCompliances_to_enbaleUPIMerchantecosystem_0.pdf)

**Validations**
The group or inidividuals involved in settling up the expenses should reside in India.

**Edge case:**

_Payment status:_

For mobile app, there could be a way to know the payment status and mark the status as settled up in our system.
But when a person scans the QR code from web and completes the transaction, there is no way to know the status because the transaction started outside the device.
A payment gateway integration can solve this. For the scope of this issue, we can keep the flow similar to cash transaction and rely on user confirmation when using QR code (web).

**Comment by KMKoushik** [comment]
@khitab we should think about how we can easily other providers for other contries, 

it should be in a way where we can add multiple providers as link which users can be clicked while trying to settle up, this way we can support any providers that allows link sharing.

**Comment by khitab** [comment]
On searching for popular P2P payment solutions, found that payment links / pages are not the first class citizens in the suite.

Examples:

- Google Pay had P2P payments in the US but it is [discontinued](https://support.google.com/googlepay/answer/14555219?hl=en).
- Apple has [P2P payments](https://support.apple.com/en-us/105013) but it does not have link feature.
- Paypal has [PayPal.me](https://www.paypal.com/us/digital-wallet/send-receive-money/paypal-me).

There are payment solutions like Venmo, Zelle etc that may have payment links. Having a payment link functionality for settling up can be a generic solution. However, UPI will need a different implementation. UPI, being a DPG, works on an open specification. Just like Email, UPI has VPA and payments can be made using any app. The number of countries supporting [UPI is growing](https://www.npci.org.in/who-we-are/group-companies/npci-international/list-of-countries).

We can create another Github issue for including support for payment links.

**Comment by krokosik** [comment]
Neither of the linked UPI sites works for me and I have never heard of this standard. What is more, this issue is very open, without an implementation plan and until it becomes more clear to me how this could be implemented, I am moving this to discussions.

**Reply by krokosik** [reply]
I've read some more on what UPI is and it's an India only standard. Since I am not from India there is no way for me to maintain it or even test it.
